src/coredump/coredump.c

```diff
@@ -1628,42 +1664,56 @@ static int forward_coredump_to_container(Context *context) {
         return 0;
 }
 
-static int gather_pid_mount_tree_fd(const Context *context) {
-        _cleanup_close_ int mntns_fd = -EBADF, root_fd = -EBADF;
+static int gather_pid_mount_tree_fd(const Context *context, int *ret_fd) {
+        _cleanup_close_ int mntns_fd = -EBADF, root_fd = -EBADF, fd = -EBADF;
         _cleanup_close_pair_ int pair[2] = EBADF_PAIR;
-        int fd = -EBADF, r;
-        pid_t child;
+        int r;
 
         assert(context);
+        assert(ret_fd);
 
         /* Don't bother preparing environment if we can't pass it to libdwfl. */
 #if !HAVE_DWFL_SET_SYSROOT
-        return -EBADF;
+        *ret_fd = -EOPNOTSUPP;
+        return 0;
 #endif
 
-        if (!arg_access_container)
-                return -EBADF;
+        if (!arg_enter_namespace) {
+                *ret_fd = -EHOSTDOWN;
+                return 0;
+        }
 
         if (socketpair(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0, pair) < 0)
                 return log_error_errno(errno, "Failed to create socket pair: %m");
 
-        r = namespace_open(context->pid,  NULL, &mntns_fd, NULL, NULL, &root_fd);
+        r = namespace_open(context->pid, NULL, &mntns_fd, NULL, NULL, &root_fd);
```

Comment:
would be nice to prefix the `NULL` parameters with their argument names, i.e. `/* pidns_fd= */` and so on. You know the drill